app/src/main/java/org/p2p/wallet/moonpay/ui/new/NewBuyPresenter.kt

```diff
@@ -0,0 +1,426 @@
+    }
+
+    private fun loadAvailablePaymentMethods() {
+        launch {
+            // show loading
+            currentAlphaCode = paymentMethodsInteractor.getBankTransferAlphaCode()
+            val availablePaymentMethods = paymentMethodsInteractor.getAvailablePaymentMethods(currentAlphaCode)
+            selectedPaymentMethod = availablePaymentMethods.first { it.isSelected }
+            paymentMethods.addAll(availablePaymentMethods)
+            view?.showPaymentMethods(paymentMethods)
+            validatePaymentMethod()
+
+            preselectMinimalFiatAmount()
+        }
+    }
+
+    private fun loadMoonpayBuyQuotes() {
+        launch {
+            val currencyCodes = currenciesToSelect.map { it.code }
+            moonpayBuyInteractor.loadQuotes(currencyCodes, tokensToBuy)
+        }
+    }
+
+    private fun preselectMinimalFiatAmount() {
+        view?.showPreselectedAmount(DEFAULT_MIN_AMOUNT.toString())
+        setBuyAmount(DEFAULT_MIN_AMOUNT.toString(), isDelayEnabled = false)
+    }
+
+    override fun onBackPressed() {
+        buyAnalytics.logBuyGoingBack(
+            buySum = currentBuyViewData?.receiveAmount?.toBigDecimal() ?: BigDecimal.ZERO,
+            buyCurrency = selectedToken.tokenSymbol,
+            buyUSD = currentBuyViewData?.price ?: BigDecimal.ZERO
+        )
+        view?.close()
+    }
+
+    override fun onPaymentMethodSelected(selectedMethod: PaymentMethod) {
+        selectedPaymentMethod = selectedMethod
+        paymentMethods.forEach { paymentMethod ->
+            paymentMethod.isSelected = paymentMethod.method == selectedMethod.method
+        }
+        validatePaymentMethod()
+
+        view?.showPaymentMethods(paymentMethods)
+    }
+
+    private fun validatePaymentMethod() {
+        if (selectedPaymentMethod.method == Method.BANK_TRANSFER) {
+            if (currentAlphaCode == BANK_TRANSFER_UK_CODE) {
+                selectCurrency(BuyCurrency.Currency.create(Constants.GBP_SYMBOL))
+            } else {
+                selectCurrency(BuyCurrency.Currency.create(Constants.EUR_SYMBOL))
+            }
+        }
+    }
+
+    override fun onSelectTokenClicked() {
+        moonpayBuyInteractor.getQuotesByCurrency(selectedCurrency.code).forEach { quote ->
+            tokensToBuy.find { it.tokenSymbol == quote.token.tokenSymbol }?.let {
+                it.rate = quote.price
+                it.currency = quote.currency
+            }
+        }
+        view?.showTokensToBuy(selectedToken, tokensToBuy)
+    }
+
+    override fun onSelectCurrencyClicked() {
+        if (currencySelectionEnabled) {
+            view?.showCurrency(currenciesToSelect, selectedCurrency)
+        }
+    }
+
+    override fun onTotalClicked() {
+        buyDetailsState?.let {
+            view?.showDetailsBottomSheet(it)
+        }
+    }
+
+    override fun setToken(token: Token) {
+        selectedToken = token
+        recalculate()
+    }
+
+    private fun selectCurrency(currency: BuyCurrency.Currency) {
```

Comment:
what's the difference between setCurrency and selectCurrency? Looks like both are doing the same job according to the name

Reply:
@kabramovp2p